csharp/test/Drivers/Apache/Hive2/DecimalUtilityTests.cs

```diff
@@ -0,0 +1,173 @@
+﻿/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Data.SqlTypes;
+using System.Diagnostics;
+using System.Globalization;
+using Apache.Arrow.Adbc.Drivers.Apache.Hive2;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Apache.Arrow.Adbc.Tests.Drivers.Apache.Hive2
+{
+    /// <summary>
+    /// Class for testing the Decimal Utilities tests.
+    /// </summary>
+    public class DecimalUtilityTests(ITestOutputHelper outputHelper)
+    {
+        private readonly ITestOutputHelper _outputHelper = outputHelper;
+
+        [SkippableTheory]
+        [MemberData(nameof(Decimal128Data))]
+        public void TestCanConvertDecimal(string value, int precision, int scale, int byteWidth, byte[] expected, SqlDecimal? expectedDecimal = default)
+        {
+            byte[] actual = new byte[byteWidth];
+            DecimalUtility.GetBytes(value, precision, scale, byteWidth, actual);
+            Assert.Equal(expected, actual);
+            Assert.Equal(0, byteWidth % 4);
+            int[] buffer = new int[byteWidth / 4];
+            for (int i = 0; i < buffer.Length; i++)
+            {
+                buffer[i] = BitConverter.ToInt32(actual, i * sizeof(int));
+            }
+            SqlDecimal actualDecimal = GetSqlDecimal128(actual, 0, precision, scale);
+            if (expectedDecimal != null) Assert.Equal(expectedDecimal, actualDecimal);
+        }
+
+        [Fact(Skip = "Run manually to confirm equivalent performance")]
+        public void TestConvertDecimalPerformance()
+        {
+            Stopwatch stopwatch = new();
```

Comment:
As an aside, I highly recommend Benchmark.NET for any kind of performance testing.